packages/rabbitmq/src/rabbitmq.interfaces.ts

```diff
@@ -45,8 +45,16 @@ export interface MessageHandlerOptions {
   queue?: string;
   queueOptions?: QueueOptions;
   errorBehavior?: MessageHandlerErrorBehavior;
+  errorCallbacks?: IMessageErrorCallback[];
+  allowNonJsonMessages?: boolean;
 }
 
+export type IMessageErrorCallback = (
```

Comment:
The signature should be this:
```
IMessageErrorCallback = (
      connection: AmqpConnection,
      channel: amqplib.Channel,
      msg: amqplib.ConsumeMessage,
      error: any
    ) => void | Promise<void> 
```
so that the user has access to the AmqpConnection instance when they're implementing error handling. That way they can optionally publish messages or do other more advanced stuff.

I think we should restrict return to `void | Promise<void>` to prevent them from trying to return a value from the error handling function because nothing would be done with the return object